xarray/core/npcompat.py

```diff
@@ -28,14 +28,48 @@
 # THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-from typing import TYPE_CHECKING, Any, Literal, Sequence, TypeVar, Union
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    List,
+    Literal,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+)
 
 import numpy as np
 from packaging.version import Version
 
 # Type annotations stubs
 try:
     from numpy.typing import ArrayLike, DTypeLike
+    from numpy.typing._dtype_like import _DTypeLikeNested, _ShapeLike, _SupportsDType
+
+    # Xarray requires a Mapping[Hashable, dtype] in many places which
+    # conflics with numpys own DTypeLik (with dtypes for fields).
+    # This is a copy of this DTypeLike that allows only non-Mapping dtypes.
```

Comment:
Excellent, thanks! It can also be helpful to add a link to the numpy code; not strictly required tho